packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -158,6 +198,78 @@ const callbackIfFileIncluded = (meta: Metadata, next: Metadata) => {
   }
 };
 
+/**
+ * Extracts CSS data from a conditional expression node.
+ * Eg. props.isPrimary && props.isBolded ? ({ color: 'blue' }) : ({ color: 'red'})
+ *
+ * @param node Node we're interested in extracting CSS from.
+ * @param state Babel state - should house options and meta data used during the transformation.
+ */
+const extractConditionalExpression = (node: t.ConditionalExpression, meta: Metadata): CSSOutput => {
+  const variables: CSSOutput['variables'] = [];
+  const css: CSSOutput['css'] = [];
+  let logicalItem: LogicalCssItem[] = [];
+
+  if (t.isObjectExpression(node.consequent)) {
+    const consequent = buildCss(node.consequent, meta);
+    logicalItem = getLogicalItemFromConditionalExpression(consequent.css, node, 'consequent');
+
+    css.push(...logicalItem);
+    variables.push(...consequent.variables);
+  } else if (t.isIdentifier(node.consequent)) {
+    const { value: interpolation, meta: updatedMeta } = evaluateExpression(node.consequent, meta);
```

Comment:
Added extra logic in https://github.com/atlassian-labs/compiled/pull/788/commits/6f7909ecb738050d8876735d79ef75c684eb067b to deal with Mixins